### PR TITLE
share location status modal

### LINKED ISSUE #287: [frontend][web] Allow the user to share the stats of his postal code
This feature allows the user to share the stats of his postal code in social networks.

Dashboard with a share button and a few modifications
https://www.figma.com/file/BfK7HtPsvxZAwZLg1ebz2u/Covidografia?node-id=3002%3A7007

It's not only the share button. The postal code name is also presented. Also, we now present how old is the last Actualização de estado (see #120)

When the user clicks the "Partilhar estado da minha zona", a dialog is open:
https://www.figma.com/file/BfK7HtPsvxZAwZLg1ebz2u/Covidografia?node-id=3105%3A5224

The URL of the image is returned in the user object. See #122

Requires #120, #121, #122, #123, #124

### LINKED ISSUE #357: [frontend] After login in with Google, the loginmethod = Facebook
The loginmethod is being set to Facebook after login in with Google.

The test was preceded with complete wipe of cookies and localstorage.

### PR DESCRIPTION
**Pull request by diogoteix**
WIP Reason: missing backend

### CODE REVIEW

File: Trackovid19-web/src/app/helpers/share-helper.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      FB.ui(
+        {
+          display: 'popup',
+          method: 'share',
+          href: link,
+          quote: quote || null,
+          hashtag: (hashtag && `#${hashtag}`) || null,
+        },
+        response => {
+          console.log(response);
```
Comment (nunofgm) [inline Trackovid19-web/src/app/helpers/share-helper.ts]: don't need this

File: Trackovid19-web/src/app/helpers/share-helper.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  facebookMessenger(text: string) {
+    if (typeof FB !== 'undefined' && FB != null) {
+      FB.ui(
+        {
+          method: 'send',
+          link: `${text}`,
+        },
+        response => {
+          console.log(response);
```
Comment (nunofgm) [inline Trackovid19-web/src/app/helpers/share-helper.ts]: same here

File: Trackovid19-web/src/app/screens/main/main.component.scss [hunk 3]
```diff
@@ -86,6 +86,27 @@
...
+  display: block;
+  box-sizing: border-box;
+  font-weight: bold;
+  border-radius: 2.5rem;
+  outline: none;
+  cursor: pointer;
+  padding: 1.125rem 2.5rem;
+  margin: 0 12px;
+  border: 1px solid #fa6400;
+  font-family: Archivo;
```
Comment (nunofgm) [inline Trackovid19-web/src/app/screens/main/main.component.scss]: I think @fabioscsilva  already removed this font-family right?

**Review by nunofgm — APPROVED** [review]

File: Trackovid19-web/src/app/screens/main/main.component.scss (hunk 3 above)
Comment (fabioscsilva) [reply Trackovid19-web/src/app/screens/main/main.component.scss]: Yes, this is not need. We have it on body

File: Trackovid19-web/src/app/shared/share-location-status/share-locations-status-modal.component.html [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        <img src="assets/img/icons/fb-share-icon.svg" />
+      </div>
+
+      <!-- Whatsapp -->
+      <div id="whatsapp" class="icon-share" (click)="shareWhatsapp()">
+        <img src="assets/img/icons/whatsapp-share-icon.svg" />
+      </div>
+
+      <!-- Facebook messenger -->
+      <!-- TODO: This needs to be reviewed -->
```
Comment (fabioscsilva) [inline Trackovid19-web/src/app/shared/share-location-status/share-locations-status-modal.component.html]: Is missing something?

**Review by fabioscsilva — APPROVED** [review]

File: Trackovid19-web/src/app/helpers/share-helper.ts (hunk 1 above)
Comment (diogoteix) [reply Trackovid19-web/src/app/helpers/share-helper.ts]: Done!

File: Trackovid19-web/src/app/helpers/share-helper.ts (hunk 2 above)
Comment (diogoteix) [reply Trackovid19-web/src/app/helpers/share-helper.ts]: Done!

File: Trackovid19-web/src/app/screens/main/main.component.scss (hunk 3 above)
Comment (diogoteix) [reply Trackovid19-web/src/app/screens/main/main.component.scss]: Done!

File: Trackovid19-web/src/app/shared/share-location-status/share-locations-status-modal.component.html (hunk 4 above)
Comment (diogoteix) [reply Trackovid19-web/src/app/shared/share-location-status/share-locations-status-modal.component.html]: Nop!

**Pull request merged**